inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/RedisSource.java

```diff
@@ -84,6 +181,278 @@ public boolean sourceFinish() {
+        redisReplicator.addCommandParser(CommandName.name("BITFIELD"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("SETBIT"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("SREM"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("UNLINK"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("SWAPDB"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("MULTI"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("EXEC"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("ZREMRANGEBYSCORE"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("ZREMRANGEBYRANK"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("ZREMRANGEBYLEX"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("LTRIM"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("SORT"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("RPOPLPUSH"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("ZPOPMIN"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("ZPOPMAX"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("REPLCONF"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("XACK"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("XADD"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("XCLAIM"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("XDEL"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("XGROUP"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("XTRIM"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("XSETID"), defaultCommandParser);
+        // since redis 6.2
+        redisReplicator.addCommandParser(CommandName.name("COPY"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("LMOVE"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("BLMOVE"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("ZDIFFSTORE"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("GEOSEARCHSTORE"), defaultCommandParser);
+        // since redis 7.0
+        redisReplicator.addCommandParser(CommandName.name("SPUBLISH"), defaultCommandParser);
+        redisReplicator.addCommandParser(CommandName.name("FUNCTION"), defaultCommandParser);
+        // add EventListener
+        redisReplicator.addEventListener((replicator, event) -> {
```

Comment:
Sorry, can we only obtain data by subscribing to Redis messages? Suggest setting Redis command to obtain data

Reply:
Sorry, do you mean that EventListener will only obtain redis data that is added after startup ? when RedisReplicator connect to redis server it will first synchronizes the rdb file, then PostRdbSyncEvent event is triggered which means historical data of Redis is synchronized, then the redisReplicator switches to command incremental synchronization mode and captures every command that occurs in the Redis instance (such as SET, DEL, etc.) and synchronizes it

Reply:
I hope we can also specify Redis commands to collect the data we want, rather than just synchronizing all Redis data.

Reply:
got it, maybe i didn't do that in the right way

Reply:
> I hope we can also specify Redis commands to collect the data we want, rather than just synchronizing all Redis data.

Is it a good idea to use Jedis on top of the redis replicator to operate and filter out the data we need ?

Reply:
> > I hope we can also specify Redis commands to collect the data we want, rather than just synchronizing all Redis data.
> 
> Is it a good idea to use Jedis on top of the redis replicator to operate and filter out the data we need ?

yes

Reply:
ok, i will fix it